File: CodeEval/Easy/the_major_element/the_major_element.js
var fs = require("fs");                                                                                   ;

var elements = [];
var el_length,
    dict,
    result;

fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
  if (line !== '') {
    elements = line.split(',');
    el_length = Math.floor(elements.length/2);
    dict = elements.reduce(function (map, num) {
        map[num] = (map[num] || 0) + 1;
        return map;
    }, Object.create(null));
    result = Object.keys(dict).filter(function(el) {
        return dict[el] > el_length
    });
    console.log(result[0] === undefined ? "None": result.toString());
  }

});

File: CodeEval/Easy/the_major_element/the_major_element.rb
File.open(ARGV[0]).each_line do |line|
    result = Hash.new(0)
    elements = line.chomp.split(',')
    el_length = elements.length/2
    elements.each {|num| result[num] += 1 }
    res = result.find { |k, v| v > el_length }
    puts res ? res[0] : "None"
end
